packages/flutter/test/cupertino/dialog_test.dart

```diff
@@ -132,28 +132,26 @@ void main() {
     semantics.dispose();
   });
 
-  testWidgets('Dialog default action styles', (WidgetTester tester) async {
+  testWidgets('Dialog default action style', (WidgetTester tester) async {
     await tester.pumpWidget(boilerplate(const CupertinoDialogAction(
       isDefaultAction: true,
       child: Text('Ok'),
     )));
 
     final DefaultTextStyle widget = tester.widget(find.byType(DefaultTextStyle));
 
-    expect(widget.style.fontWeight, equals(FontWeight.w400));
+    expect(widget.style.fontWeight, equals(FontWeight.bold));
   });
 
-  testWidgets('Default and destructive style', (WidgetTester tester) async {
+  testWidgets('Dialog destructive action style', (WidgetTester tester) async {
     await tester.pumpWidget(boilerplate(const CupertinoDialogAction(
-      isDefaultAction: true,
       isDestructiveAction: true,
       child: Text('Ok'),
     )));
 
     final DefaultTextStyle widget = tester.widget(find.byType(DefaultTextStyle));
 
-    expect(widget.style.fontWeight, equals(FontWeight.w400));
-    expect(widget.style.color.red, greaterThan(widget.style.color.blue));
+    expect(widget.style.color.withOpacity(1), equals(CupertinoColors.destructiveRed));
```

Comment:
Shouldn't the `widget.style.color` be exactly the same as `CupertinoColors.destructiveRed`? Or, if the widget is internally reducing the opacity, then can we match something like `CupertinoColors.destructiveRed.withOpacity(...)`

Reply:
Since `onPressed` is not set the action button is not `enabled` and the text opacity is reduced.

https://github.com/flutter/flutter/blob/cda5dfbf03ca6534b47ddd1b23c20b54a68a2fe1/packages/flutter/lib/src/cupertino/dialog.dart#L1154-L1156

If you prefer I can flip the test to `expect(widget.style.color, equals(CupertinoColors.destructiveRed.withOpacity(0.5)))`, also, if it's possible, can be a good idea add a comment explaining why the opacity of the destructiveRed is 50% in the test.

Reply:
Yeah, I think it would be good to explicitly use the actual opacity as you suggested.

Also, if such tests do not exist, can you add 1 or 2 tests that test the opacity of the destructive button when it's enabled and when it's disabled?

Reply:
@matthew-carroll I've tried to add a specify test for the opacity when the dialog action is disabled, but the opacity doesn't correspond to `0.5`. Can't figure out why.

Reply:
It looked like there was a followup comment to this about not being able to get opacity working in the test. I'm not sure why GitHub isn't showing me that comment. If opacity isn't working as expected then I'd recommend adding a TODO stating the problem and requesting it be fixed. Then we can take care of it later.

Does that sound good to you @xster and @LongCatIsLooong ?

Reply:
@matthew-carroll I deleted it, the opacity that I get from the color isn’t exact 0.5 because its encoded in the range [0,255], but it’s a normal behavior.